flexmeasures/data/models/planning/storage.py

```diff
@@ -142,12 +144,43 @@ def compute(
                 resolution,
             )
 
-        device_constraints[0]["min"] = (soc_min - soc_at_start) * (
-            timedelta(hours=1) / resolution
+        soc_min_change = (soc_min - soc_at_start) * timedelta(hours=1) / resolution
+        soc_max_change = (soc_max - soc_at_start) * timedelta(hours=1) / resolution
+
+        if soc_minima is not None:
+            device_constraints[0]["min"] = build_device_soc_targets(
+                soc_minima,
+                soc_at_start,
+                start,
+                end,
+                resolution,
+            )
+
+        device_constraints[0]["min"] = device_constraints[0]["min"].fillna(
+            soc_min_change
+        )
+
+        if soc_maxima is not None:
+            device_constraints[0]["max"] = build_device_soc_targets(
+                soc_maxima,
+                soc_at_start,
+                start,
+                end,
+                resolution,
+            )
+
+        device_constraints[0]["max"] = device_constraints[0]["max"].fillna(
+            soc_max_change
         )
-        device_constraints[0]["max"] = (soc_max - soc_at_start) * (
-            timedelta(hours=1) / resolution
+
+        # limiting max and min to be in the range [soc_min, soc_max]
```

Comment:
I'd love to see tests for some extra cases, such as:
- some minima < min
- some maxima > max
- some minima > some maxima (during the same time slot)
- some minima > max
- some maxima < min

Maybe some of these cases don't have to involve the whole scheduler, but could just check a util function in which this clipping logic is performed. Or should they be caught by validation instead? Actually, in each case, what would be the expected / desired behaviour?